samples/transform_feedback_interleaved.html

```diff
@@ -0,0 +1,259 @@
+<html lang="en">
+
+<head>
+    <title>WebGL 2 Samples - transform_feedback_interleaved</title>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0">
+    <link rel="stylesheet" href="style.css">
+</head>
+
+<body>
+    <div id="info">WebGL 2 Samples - transform_feedback_interleaved</div>
+
+    <script id="vs-transform" type="x-shader/x-vertex">
+        #version 300 es
+        precision highp float;
+        precision highp int;
+
+        uniform mat4 MVP;
+        layout(location = 0) in vec4 position;
+
+        out vec4 color;
+
+        void main()
+        {
+            gl_Position = MVP * position;
+            color = vec4(clamp(vec2(position), 0.0, 1.0), 0.0, 1.0);
+        }
+    </script>
+
+    <!-- This fragment shader will not be called since it is used when the rasterizer is disabled. -->
+    <script id="fs-transform" type="x-shader/x-fragment">
+        #version 300 es
+        precision highp float;
+        precision highp int;
+        
+        out vec4 color;
+
+        void main()
+        {
+            color = vec4(1.0);
+        }
+    </script>
+
+    <script id="vs-feedback" type="x-shader/x-vertex">
+        #version 300 es
+        precision highp float;
+        precision highp int;
+
+        layout(location = 0) in vec4 position;
+        layout(location = 1) in vec4 color;
+
+        out vec4 v_color;
+
+        void main()
+        {
+            gl_Position = position;
+            v_color = color;
+        }
+    </script>
+
+    <script id="fs-feedback" type="x-shader/x-fragment">
+        #version 300 es
+        precision highp float;
+        precision highp int;
+
+        in vec4 v_color;
+
+        out vec4 color;
+
+        void main()
+        {
+            color = v_color;
+        }
+    </script>
+
+    <script src="utility.js"></script>
+    <script>
+    (function () {
+        'use strict';
+
+        // -- Init Canvas
+        var canvas = document.createElement('canvas');
+        canvas.width = Math.min(window.innerWidth, window.innerHeight);
+        canvas.height = canvas.width;
+        document.body.appendChild(canvas);
+
+        // -- Init WebGL Context
+        var gl = canvas.getContext('webgl2', { antialias: false });
+        if (!gl) {
+            gl = canvas.getContext('experimental-webgl2', { antialias: false });
+        }
+        var isWebGL2 = !!gl;
+        if(!isWebGL2)
+        {
+            document.getElementById('info').innerHTML = 'WebGL 2 is not available.  See <a href="https://www.khronos.org/webgl/wiki/Getting_a_WebGL_Implementation">How to get a WebGL 2 implementation</a>';
+            return;
+        }
+
+        // -- Init Program
+        var PROGRAM_TRANSFORM = 0;
+        var PROGRAM_FEEDBACK = 1;
+
+        var programs = (function(gl, vertexShaderSourceTransform, fragmentShaderSourceTransform, vertexShaderSourceFeedback, fragmentShaderSourceFeedback) {
```

Comment:
Why is this so complicated?

I get that we want to highlight the use of `transformFeedbackVaryings` in this sample so why not use the `createProgram` helper we already have for the program that isn't using transform feedback, and then cut this code in half and remove the anonymous function and make the GL calls in the main function?